### ISSUE TITLE
ValueError: MXNet to Keras: Layer weight shape (7, 7, 1, 64) not compatible with provided weight shape (7, 7, 64, 1)

### ISSUE THREAD

**Issue by emilyclaps** [microsoft/MMdnn#676]
Platform (like ubuntu 16.04/win10):
win10
Python version:
3.6
Source framework with version (like Tensorflow 1.4.1 with GPU):
MXNet
Destination framework with version (like CNTK 2.3 with GPU):
Keras 2.2.4
Pre-trained model path (webpath or webdisk path):
N/A, custom model
Running scripts:
I have an MXNet image classification model (B&W images) that was trained using SageMaker, and I'm trying to convert it to Keras to be used external to the platform. I am able to convert the model (with the .params file, symbol.json, and model-shapes.json) to IR format to Keras code snippet (.py), but when I try to convert the code snippet and weights to a .h5 file, I receive the following error: 

ValueError: Layer weight shape (7, 7, 1, 64) not compatible with provided weight shape (7, 7, 64, 1)

Please help!
---------------

**Comment by rainLiuplus** [comment]
Hi @emilyclaps,
could you upload your model for us to debug? Thanks!

**Comment by emilyclaps** [comment]
It looks like the .params file makes it too big to submit here (even when zipped)...any other way I can upload it?

**Comment by rainLiuplus** [comment]
@emilyclaps, maybe Google Drive?

**Comment by emilyclaps** [comment]
Hi @rainLiuplus - I just uploaded it to Google Drive and made the link shareable! Here you go: 
https://drive.google.com/file/d/1wwWfsTe3JN7as6QJ7HfMY_yo1z3RVBvk/view?usp=sharing

The model is in a .tar.gz format. Please let me know if you have any trouble.

**Comment by emilyclaps** [comment]
@rainLiuplus to confirm - were you able to download the model to debug? 

Thanks!!

**Comment by emilyclaps** [comment]
@rainLiuplus sorry to bother - just want to verify you were able to download the model. Otherwise any suggestions or help would be greatly appreciated! It's been a struggle to try to figure this out. 

Also - I was able to transpose the weights to be read as expected (so I think), but when I run the model it gives predictions that are the same every single time (and with nearly the exact same probability)...so I'm assuming the weights are not translated correctly. Perhaps that may provide some insight.

**Comment by rainLiuplus** [comment]
@emilyclaps sorry I am a little busy recently. The model could be downloaded and the issue is added into the work plan. BTW, maybe you could compare the output results of the original model and the converted model to determine whether the conversion is successful. Thanks!

**Comment by emilyclaps** [comment]
Yes, I did compare the results...the accuracy of the converted model is way off; it makes the same exact prediction for each image with the same probability when the original model had 90% accuracy. I was also only able to get it to work when I added 'custom objects' to the layers when load_model() is called.
